ports/raspberrypi/common-hal/pulseio/PulseIn.c

```diff
@@ -141,9 +141,16 @@ void common_hal_pulseio_pulsein_interrupt() {
 	    }
             // ignore pulses that are too short
             if (result <= MAX_PULSE && result > MIN_PULSE) {
-                self->buffer[buf_index] = (uint16_t) result;
-                buf_index++;
-                self->len++;
+              self->buffer[buf_index] = (uint16_t) result;
+	      if (self->len < self->maxlen) {
+                 self->len++;
+	        }
+	      if (buf_index < self->maxlen) {
+		 buf_index++;
+               } else {
+                 self->start = 0;
+		 buf_index = 0;
```

Comment:
Do you have pre-commit installed? I wouldn't expect this uneven indentation to pass the code formatting check.